orbit/pkg/packaging/macos_rcodesign.go

```diff
@@ -0,0 +1,80 @@
+package packaging
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/fleetdm/fleet/v4/pkg/secure"
+)
+
+func rSign(pkgPath, cert, certPwd string) error {
+	certPath := filepath.Join(os.TempDir(), "cert.p12")
+	defer os.Remove(certPath)
+	err := os.WriteFile(certPath, []byte(cert), 0o600)
+	if err != nil {
+		return fmt.Errorf("writing cert data: %e", err)
+	}
+
+	var outBuf bytes.Buffer
+	cmd := exec.Command(
+		"rcodesign",
+		"sign",
+		"--p12-file", certPath,
+		"--p12-password", certPwd,
+		pkgPath,
+	)
+	cmd.Stdout = &outBuf
+	cmd.Stderr = &outBuf
+	if err := cmd.Run(); err != nil {
+		fmt.Println(outBuf.String())
+		return fmt.Errorf("rcodesign: %w", err)
+	}
+	return nil
+}
+
+func rNotarizeStaple(pkg, apiKeyID, apiKeyIssuer, apiKeyContent string) error {
+	path, err := writeAPIKeys(apiKeyIssuer, apiKeyID, apiKeyContent)
+	defer os.Remove(path)
+	if err != nil {
+		return fmt.Errorf("writing API keys: %e", err)
+	}
+	var outBuf bytes.Buffer
+	cmd := exec.Command("rcodesign",
+		"notarize",
+		"--api-issuer", apiKeyIssuer,
+		"--api-key", apiKeyID,
+		"--staple",
+		pkg,
+	)
+	cmd.Stdout = &outBuf
+	cmd.Stderr = &outBuf
+	if err := cmd.Run(); err != nil {
+		fmt.Println(outBuf.String())
+		return fmt.Errorf("rcodesign notarize: %w", err)
+	}
+	return nil
+}
+
+func writeAPIKeys(issuer, id, content string) (string, error) {
```

Comment:
In the future (not as part of this PR) we could wrap this into a [sync.Once](https://pkg.go.dev/sync#Once). (We'll need to tune `BuildPkg` for concurrent use.)